ingestr/src/sources.py

```diff
@@ -1034,3 +1037,64 @@ def dlt_source(self, uri: str, table: str, **kwargs):
         src = asana_source()
         src.workspaces.add_filter(lambda w: w["gid"] == workspace)
         return src.with_resources(table)
+
+
+class DynamoDBSource:
+
+    AWS_ENDPOINT_PATTERN = re.compile(".*\.(.+)\.amazonaws\.com:443")
+
+    def infer_aws_region(self, uri: ParseResult) -> Optional[str]:
+        # try to infer from URI
+        matches = self.AWS_ENDPOINT_PATTERN.match(uri.netloc)
+        if matches is not None:
+            return matches[1]
+
+        # else obtain region from query string
+        region = parse_qs(uri.query).get("region")
+        if region is None:
+            return None
+        return region[0]
+    
+    def get_endpoint_url(self, url: ParseResult) -> str:
+        if self.AWS_ENDPOINT_PATTERN.match(url.netloc) is not None:
+            return f"https://{url.netloc}"
+        return url.netloc
+
+    def handles_incrementality(self) -> bool:
+        return False
+
+    def dlt_source(self, uri: str, table: str, **kwargs):
+        parsed_uri = urlparse(uri)
+
+        region = self.infer_aws_region(parsed_uri)
+        if not region:
+            raise ValueError("Region is required to connect to Dynamodb")
+
+        qs = parse_qs(quote(parsed_uri.query, safe="=&"))
+        access_key = qs.get("access_key_id")
+
+        if not access_key:
+            raise ValueError("access_key_id is required to connect to Dynamodb")
+
+        secret_key = qs.get("secret_access_key")
+        if not secret_key:
+            raise ValueError("secret_access_key is required to connect to Dynamodb")
+
+        creds = AwsCredentials(
+            aws_access_key_id=access_key[0],
+            aws_secret_access_key=TSecretStrValue(secret_key[0]),
+            region_name=region,
+            endpoint_url=self.get_endpoint_url(parsed_uri),
+        )
+
+        incremental = None
+        incremental_key = kwargs.get("incremental_key", "").strip()
+
+        if incremental_key:
+            incremental = dlt.sources.incremental(
+                incremental_key,
+                initial_value=isotime(kwargs.get("interval_start")),
+                end_value=isotime(kwargs.get("interval_end")),
```

Comment:
I don't think we should convert them to iso formats, it could be that the user is filtering for a date field instead. I'd also remove this helper function since it wouldn't be used.